src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,16 +1,216 @@
-import { Component } from 'react';
+import React, { useState } from 'react';
+import './NewMovie.scss';
 
-type Props = {};
-type State = {};
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
 
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
-
-  render() {
-    return (
-      <form>
-        Put the form here
-      </form>
-    );
-  }
+enum MovieKeys {
+  Title = 'title',
+  Description = 'description',
+  ImgUrl = 'imgUrl',
+  ImdbUrl = 'imdbUrl',
+  ImdbId = 'imdbId',
 }
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  /* ---------------Error-inputs---------------- */
+  const [titleError, setTitleError] = useState(false);
+  const [descriptionError, setDescriptionError] = useState(false);
+  const [imgUrlError, setImgUrlError] = useState(false);
+  const [imdbUrlError, setImdbUrlError] = useState(false);
+  const [imdbIdError, setImdbIdError] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setDescription('');
+    setImdbId('');
+    setImdbUrl('');
+    setImgUrl('');
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+
+    switch (name) {
+      case MovieKeys.Title:
+        setTitle(value);
+        setTitleError(false);
+        break;
+
+      case MovieKeys.Description:
+        setDescription(value);
+        setDescriptionError(false);
+        break;
+
+      case MovieKeys.ImgUrl:
+        setImgUrl(value);
+        setImgUrlError(false);
+        break;
+
+      case MovieKeys.ImdbUrl:
+        setImdbUrl(value);
+        setImdbUrlError(false);
+        break;
+
+      case MovieKeys.ImdbId:
+        setImdbId(value);
+        setImdbIdError(false);
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const handleInputBlur = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name } = event.target;
+
+    switch (name) {
+      case MovieKeys.Title:
+        if (!title) {
+          setTitleError(true);
+        }
+
+        break;
+
+      case MovieKeys.Description:
+        if (!title) {
+          setDescriptionError(true);
+        }
+
+        break;
+
+      case MovieKeys.ImgUrl:
+        if (!title) {
```

Comment:
and so on, fix it everywhere
```suggestion
        if (!imgUrl) {
```